@here/harp-mapview/lib/geometry/PhasedTileGeometryManager.ts

```diff
@@ -52,15 +52,13 @@ export class PhasedTileGeometryManager extends TileGeometryManagerBase {
     }
 
     updateTiles(tiles: Tile[]): void {
-        if (this.mapView.isDynamicFrame) {
-            this.updateSomeTiles(tiles);
-        } else {
-            this.updateAllTilesTogether(tiles);
-        }
+        let needUpdate = this.mapView.isDynamicFrame
+            ? this.updateSomeTiles(tiles)
+            : this.updateAllTilesTogether(tiles);
 
-        this.updateTileObjectVisibility(tiles);
+        needUpdate = needUpdate || this.updateTileObjectVisibility(tiles);
 
-        if (!this.checkTilesFinished(tiles)) {
+        if (needUpdate || !this.checkTilesFinished(tiles)) {
```

Comment:
```suggestion
        needUpdate = needUpdate || !this.checkTilesFinished(tiles);
        
        if (needUpdate) {
```

Reply:
Unfortunately, 
needUpdate = needUpdate || !this.checkTilesFinished(tiles);
is just as wrong as my line
if (needUpdate || !this.checkTilesFinished(tiles)) {
since checkTilesFinished() has to be called anyways.

Your suggestion makes sense, but is is a bit of a matter of personal taste. I am sticking with the shorter version, since there are only a few expressions, and readability is "good enough". It also saves a store operation (... which may be optimized away by the compiler, I know...)

Reply:
so, it would be
```suggestion
        needUpdate = needUpdate | !this.checkTilesFinished(tiles);
        
        if (needUpdate) {
```

note the |

Reply:
In the "typeless" world of javascript, this makes sense, and would work fine. Except that, for a C++ developer, it smells like a hack, since boolean !== number, and binary operators are not defined for booleans, or are they? 
"Long live static types!", and "No more truthies"!  8^)

Reply:
Another try
```suggestion
        if (!this.checkTilesFinished(tiles)) {
                needUpdate = true;
        }
        
        if (needUpdate) {
```

Reply:
If you look on booleans like on 1-bit clamped integers, then all integer operations work just fine. But ok, no more truthies.